Adapter View

Dude, I love your work man. I was using the AutoCompleteTextView, created custom adapter ext,. and working all these days.

I started using yours and love it, the only problem is my earlier AutoCompleteTextView was showing many suggestions (like 8-10) as soon as i begin typing a place name. But yours shows only 2 ... Kindly help.

@sddamico , Thanks
But when I run app, it crash because unable to find valid constructor with params Context, PlacesApi...

Here is my xml
<pre>
<com.seatgeek.placesautocomplete.PlacesAutocompleteTextView
                            android:id="@+id/places_autocomplete"
                            android:layout_width="match_parent"
                            android:layout_height="wrap_content"
                            android:background="@null"
                            android:hint="Address"
                            android:paddingBottom="5dp"
                            android:paddingTop="5dp"
                            android:singleLine="true"
                            android:textColor="@color/text_gray_4"
                            android:textColorHint="@color/text_gray_3"
                            android:textSize="15sp"
                            app:pacv_googleMapsApiKey="@string/API_server_key"
                        app:pacv_adapterClass="adapter.TestPlacesAutocompleteAdapter"/>
</pre>
Here is log.
<pre>
Caused by: android.view.InflateException: Unable to find valid constructor with params Context, PlacesApi, AutocompleteResultType, and DefaultAutocompleteHistoryManager for specified adapterClass: com.teusoft.sirved.adapter.TestPlacesAutocompleteAdapter
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.adapterForClass(PlacesAutocompleteTextView.java:225)
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.init(PlacesAutocompleteTextView.java:127)
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.<init>(PlacesAutocompleteTextView.java:78)
                                           at java.lang.reflect.Constructor.newInstance(Native Method) 
                                           at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
                                           at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
                                           at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:365) 
                                           at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:378) 
                                           at android.app.Activity.setContentView(Activity.java:2145) 
                                           at com.teusoft.sirved.ProfileActivity.onCreate(ProfileActivity.java:151) 
                                           at android.app.Activity.performCreate(Activity.java:5990) 
                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106) 
                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278) 
                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387) 
                                           at android.app.ActivityThread.access$800(ActivityThread.java:151) 
                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303) 
                                           at android.os.Handler.dispatchMessage(Handler.java:102) 
                                           at android.os.Looper.loop(Looper.java:135) 
                                           at android.app.ActivityThread.main(ActivityThread.java:5254) 
                                           at java.lang.reflect.Method.invoke(Native Method) 
                                           at java.lang.reflect.Method.invoke(Method.java:372) 
                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903) 
                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698) 
                                        Caused by: java.lang.NoSuchMethodException: <init> [class android.content.Context, class com.seatgeek.placesautocomplete.PlacesApi, class com.seatgeek.placesautocomplete.model.AutocompleteResultType, class com.seatgeek.placesautocomplete.history.DefaultAutocompleteHistoryManager]
                                           at java.lang.Class.getConstructor(Class.java:531)
                                           at java.lang.Class.getConstructor(Class.java:495)
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.adapterForClass(PlacesAutocompleteTextView.java:216)
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.init(PlacesAutocompleteTextView.java:127) 
                                           at com.seatgeek.placesautocomplete.PlacesAutocompleteTextView.<init>(PlacesAutocompleteTextView.java:78) 
                                           at java.lang.reflect.Constructor.newInstance(Native Method) 
                                           at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
                                           at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
                                           at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
                                           at android.view.LayoutInflater.inflate(LayoutInflater.java:365) 
                                           at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:378) 
                                           at android.app.Activity.setContentView(Activity.java:2145) 
                                           at com.teusoft.sirved.ProfileActivity.onCreate(ProfileActivity.java:151) 
                                           at android.app.Activity.performCreate(Activity.java:5990) 
                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106) 
                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278) 
                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387) 
                                           at android.app.ActivityThread.access$800(ActivityThread.java:151) 
                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303) 
                                           at android.os.Handler.dispatchMessage(Handler.java:102) 
                                           at android.os.Looper.loop(Looper.java:135) 
                                           at android.app.ActivityThread.main(ActivityThread.java:5254) 
                                           at java.lang.reflect.Method.invoke(Native Method) 
                                           at java.lang.reflect.Method.invoke(Method.java:372) 
                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903) 
                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698) 
</pre>

--- a/demo-app/src/main/res/layout/activity_places_autocomplete.xml
+++ b/demo-app/src/main/res/layout/activity_places_autocomplete.xml
@@ -15,7 +15,7 @@
         android:layout_height="wrap_content"
         android:layout_gravity="center_horizontal"
         android:padding="16dp"
-        app:pacv_languageCode="ru"
+        app:pacv_languageCode="en"
         app:pacv_googleMapsApiKey="@string/google_maps_api_key"
         app:pacv_adapterClass="com.seatgeek.placesautocompletedemo.TestPlacesAutocompleteAdapter"/>
 

--- a/placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/PlacesAutocompleteTextView.java
+++ b/placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/PlacesAutocompleteTextView.java
@@ -230,7 +230,7 @@ public class PlacesAutocompleteTextView extends AutoCompleteTextView {
                     ", " +
                     AutocompleteResultType.class.getSimpleName() +
                     ", and " +
-                    DefaultAutocompleteHistoryManager.class.getSimpleName() +
+                    AutocompleteHistoryManager.class.getSimpleName() +
                     " for specified adapterClass: " + adapterClass, e);
         }
 

--- a/placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/adapter/AbstractPlacesAutocompleteAdapter.java
+++ b/placesautocomplete/src/main/java/com/seatgeek/placesautocomplete/adapter/AbstractPlacesAutocompleteAdapter.java
@@ -26,10 +26,10 @@ public abstract class AbstractPlacesAutocompleteAdapter extends ArrayAdapter<Pla
     @NonNull
     private final PlacesApiFilter mFilter;
 
-    protected AbstractPlacesAutocompleteAdapter(@NonNull final Context context,
-                                                @NonNull final PlacesApi api,
-                                                @Nullable final AutocompleteResultType autocompleteResultType,
-                                                @Nullable AutocompleteHistoryManager history) {
+    public AbstractPlacesAutocompleteAdapter(@NonNull final Context context,
+                                             @NonNull final PlacesApi api,
+                                             @Nullable final AutocompleteResultType autocompleteResultType,
+                                             @Nullable final AutocompleteHistoryManager history) {
         super(context, 0);
 
         mFilter = new PlacesApiFilter(

